app/api/routes/product.py

```diff
@@ -0,0 +1,98 @@
+from datetime import datetime
+from typing import List
+from uuid import UUID
+from fastapi import APIRouter, HTTPException, Depends, status
+from sqlmodel import Session
+from app.api.dependencies import get_current_admin
+from app.database import get_session
+
+from app.models import User
+from app.schemas.product_schema import CreateProductRequest, Product, UpdateProductRequest
+from app.services.product_services import ProductService
+
+router = APIRouter()
+
+# Create product (only admin)
+@router.post("/products", response_model=Product, status_code=status.HTTP_201_CREATED)
+async def create_product(
+    product: CreateProductRequest,
+    current_admin: User = Depends(get_current_admin),
+    session: Session = Depends(get_session)
+):
+    product_service = ProductService(session)
+    try:
+        return product_service.create_product(product)
+    except HTTPException as http_exc:
```

Comment:
No need to catch the exception if you want to pass it as is, because it will propagate all the way to the end of your program. You can delete this line and see what I mean.